language/transaction-builder-generator/src/java.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::common::type_not_allowed;
+use libra_types::transaction::{ArgumentABI, ScriptABI, TypeArgumentABI};
+use move_core_types::language_storage::TypeTag;
+
+use std::{
+    io::{Result, Write},
+    path::PathBuf,
+};
+
+/// Output transaction builders in Java for the given ABIs.
+pub fn output(
+    out: &mut dyn Write,
+    abis: &[ScriptABI],
+    package: Option<&str>,
+    class_name: &str,
+) -> Result<()> {
+    output_preamble(out, package)?;
+    writeln!(out, "\npublic final class {} {{\n", class_name)?;
+    for abi in abis {
+        output_builder(out, abi)?;
+    }
+    writeln!(out, "\n}}\n")?;
+    Ok(())
+}
+
+fn output_preamble(out: &mut dyn Write, package: Option<&str>) -> Result<()> {
+    if let Some(name) = package {
+        writeln!(out, "package {};\n", name)?;
+    }
+    writeln!(
+        out,
+        r#"
+import java.math.BigInteger;
+import java.util.Arrays;
+import libra.types.AccountAddress;
```

Comment:
I suggest we use package org.libra.something  here

Reply:
`org.libra.types` and `org.libra.lcs` make sense but then it gets tricky with `bincode` and `serde`. I need to think about it.

Reply:
Yes, I think I can work with `org.libra` and `com.facebook`.

Reply:
@thefallentree addressed